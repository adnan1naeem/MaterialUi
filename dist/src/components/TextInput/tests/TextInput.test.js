"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = require("../");
const react_test_renderer_1 = __importDefault(require("react-test-renderer"));
const react_1 = __importDefault(require("react"));
require("jest");
function handleTextInputChange() {
    console.log('I am clicked');
}
test('TextInput component with default values', () => {
    const textInput = react_test_renderer_1.default.create(react_1.default.createElement(__1.TextInput, { placeholder: "hello", defaultValue: "nooo", onChange: handleTextInputChange, autoComplete: "Hello World !!", autoFocus: true })).toJSON();
    expect(textInput.props.style).toEqual(undefined);
    expect(textInput).toMatchSnapshot();
});
test('TextInput component with editable true', () => {
    const textInput = react_test_renderer_1.default.create(react_1.default.createElement(__1.TextInput, { placeholder: "hello", editable: false, defaultValue: "nooo", onChange: handleTextInputChange, autoComplete: "Hello World !!", autoFocus: true })).toJSON();
    expect(textInput.props.editable).toBe(false);
    expect(textInput).toMatchSnapshot();
});
test('TextInput component with onChangeText', () => {
    const textInput = react_test_renderer_1.default.create(react_1.default.createElement(__1.TextInput, { placeholder: "hello", editable: false, defaultValue: "nooo", onChangeText: handleTextInputChange, autoComplete: "Hello World !!", autoFocus: true })).toJSON();
    expect(textInput.props.onChangeText).toEqual(undefined);
    expect(textInput).toMatchSnapshot();
});
//# sourceMappingURL=TextInput.test.js.map