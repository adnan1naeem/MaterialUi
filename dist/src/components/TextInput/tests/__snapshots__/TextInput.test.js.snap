// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component with default values 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1"
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
  style={undefined}
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-11"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <input
      aria-invalid={false}
      aria-required={false}
      autoComplete="Hello World !!"
      autoFocus={true}
      className="MuiInput-input-14 MuiInput-inputSingleline-17"
      defaultValue="nooo"
      disabled={false}
      id={undefined}
      name={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="hello"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
</div>
`;

exports[`TextInput component with editable true 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1"
  editable={false}
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
  style={undefined}
>
  <div
    className="MuiInput-root-5 MuiInput-disabled-10 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-11"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <input
      aria-invalid={false}
      aria-required={false}
      autoComplete="Hello World !!"
      autoFocus={true}
      className="MuiInput-input-14 MuiInput-inputDisabled-16 MuiInput-inputSingleline-17"
      defaultValue="nooo"
      disabled={true}
      id={undefined}
      name={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="hello"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
</div>
`;

exports[`TextInput component with onChangeText 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-1"
  editable={false}
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
  style={undefined}
>
  <div
    className="MuiInput-root-5 MuiInput-disabled-10 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-11"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <input
      aria-invalid={false}
      aria-required={false}
      autoComplete="Hello World !!"
      autoFocus={true}
      className="MuiInput-input-14 MuiInput-inputDisabled-16 MuiInput-inputSingleline-17"
      defaultValue="nooo"
      disabled={true}
      id={undefined}
      name={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="hello"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
</div>
`;
